src/sampling/adaptive.jl

```diff
@@ -0,0 +1,204 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+using DataStructures: BinaryHeap
+using LinearAlgebra: norm
+import IterTools
+
+"""
+    AdaptiveSampling(; [options])
+
+Incrementally sample points such that the error from linear interpolation between these points is small. This is useful for plotting parametric functions or Bezier curves where curvature is very non-uniform.
+
+This is *only* implemented for geometries with a one-dimensional parameter domain (i.e., realistically, [`ParametrizedCurve`](@ref) or [`BezierCurve`](@ref))!
+
```

Comment:
```suggestion
```

No need to make this information available in the docstring. At some point in the future, we hope to fill in the gaps.